pkg/yang/types.go

```diff
@@ -280,7 +280,8 @@ check:
 		y.IdentityBase = resolvedBase.Identity
 	}
 
-	if t.Range != nil {
+	// If parsing fractionDigits fails, the ranges should not be parsed
+	if t.Range != nil && isDecimal64 && y.FractionDigits != 0 {
```

Comment:
Could you check that the logic here is right? I think integer types may also have a range that isn't `nil`, but this resolution of the range is now only implemented for `decimal64`. If we could add a testcase to ensure that this is properly covered (and demonstrate that we didn't break this) that'd be great.

Reply:
You're right, parsing the range of integer types is ignored, I've adjusted the logic and added a testcase.